kuma/dashboards/jinja2/dashboards/spam.html

```diff
@@ -0,0 +1,95 @@
+                 ('published_spam', 'Published Spam', 'data'),
+                 ('blocked_spam', 'Blocked Spam', 'data'),
+                 ('blocked_ham', 'Blocked Ham', 'data'),
+                 ('true_positive_rate', 'True Positive Rate', 'rate'),
+                 ('true_negative_rate', 'True Negative Rate', 'rate')) %}
+{% block content %}
+<h1>{{ title }}</h1>
+{% if improperly_configured %}
+  <p>{{ improperly_configured }}</p>
+{% endif %}
+{% if processing %}
+  <p>The report is being processed. Please reload in a minute.</p>
+{% else %}
+
+  <table class="spam-trends-table">
+    <caption>{{ _('Trends Over Time, Ending %(date)s', date=end_date ) }}</caption>
+    <thead>
+      <tr>
+        <th class="stat-header">Period</th>
+        <th class="stat-header">Start</th>
+        {% for stat, stat_name, stat_class in stat_ids -%}
+        <th class="stat-header">{{ stat_name }}</th>
+        {% endfor -%}
+      </tr>
+    </thead>
+    <tbody>
+      {% for period in trends -%}
+      <tr>
+        <th scope="row">{{ period.name }}</th>
+        <th scope="row">{{ period.start }}</th>
+        {%- for stat, stat_name, stat_class in stat_ids %}
+        {%- if stat_class == 'rate' %}
+        <td class="{{ stat_class }}">{{ "%0.1f%%" | format(100 * period.stats.get(stat, 0.0)) }}</td>
+        {%- elif stat_class == 'float' %}
+        <td class="{{ stat_class }}">{{ "%0.1f" | format(period.stats.get(stat, 0.0)) }}</td>
+        {%- else %}
+        <td class="{{ stat_class }}">{{ period.stats[stat] | default(0) }}</td>
+        {%- endif %}
+        {%- endfor %}
+      </tr>
+      {% endfor -%}
+    </tbody>
+  </table>
+
+  {% if recent_spam %}
+  <table class="spam-events-table">
+    <caption>Published Spam</caption>
+    <colgroup>
+      <col span="1" class="data" />
+      <col span="1" class="data" />
+      <col span="1" class="data" />
+      <col span="1" class="data" />
+      <col span="1" class="name" />
+      <col span="1" class="path" />
+    </colgroup>
+    <thead>
+      <tr>
+        <th class="data">Date</th>
+        <th class="data">Time Active (seconds)</th>
+        <th class="data">Estimated Viewers</th>
+        <th class="data">Revision</th>
+        <th class="name">Change Type</th>
+        <th class="path">Path</th>
+      </tr>
+    </thead>
+    <tbody>
+      {% for rs in recent_spam -%}
+      <tr>
+        <td class="data">{{ rs.date }}</td>
+        <td class="data">{{ rs.time_active }}</td>
+        <td class="data">{{ rs.viewers }}</td>
+        <td class="data"><a href="{{ rs.revision_path }}">{{ rs.revision_id }}</a></td>
+        <td class="name">{{ rs.change_type }}</td>
+        <td class="path"><a href="{{ rs.document_path }}">{{ rs.document_path }}</a></td>
+      </tr>
+      {% endfor -%}
+    </tbody>
+  </table>
+  {% else %}
+  <p>No recent published spam.</p>
+  {% endif %}{# recent spam #}
+  <ul>
+    <li>Recent event data generated at {{ now }}</li>
```

Comment:
Either works, since both utility functions involved generated their own timestamp of when they were run -- `now` for the recent spam events, `generated` for the spam stats.  Checking this did make me notice, however, that I have both timestamps being generated in different ways: one with Django's `timezone.now()` and the other one with `datetime.datetime.now()`.